### PR TITLE
Mention the risk of filling up the disk in the security considerations.

### LINKED ISSUE #277: Mention something about filling up the disk/problems with writing very large files
As mentioned in w3ctag/design-reviews#600, we should mention something in the security considerations section regarding consequences of writing very far ahead, and in general the ability to fill up a users storage by writing to files that are not part of a websites storage quota.

### PR DESCRIPTION
**Pull request by mkruisselbrink**
Also be explicit in the "write" and "truncate" steps that these
operations can fail with a QuotaExceededError if the file would be too
big.

Fixes #277

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/file-system-access/pull/278.html" title="Last updated on Feb 18, 2021, 3:25 AM UTC (205cfe2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/file-system-access/278/edc4d88...205cfe2.html" title="Last updated on Feb 18, 2021, 3:25 AM UTC (205cfe2)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -999,6 +999,13 @@ runs these steps:
          1. Let |tail| be a [=byte sequence=] containing the last
             |oldSize| - (|writePosition| + |data|.[=byte sequence/length=]) bytes of |stream|.[=[[buffer]]=].
       1. Set |stream|.[=[[buffer]]=] to the concatenation of |head|, |data| and |tail|.
+      1. If the operations modifying |stream|.[=[[buffer]]=] in the previous steps failed
+         due to exceeding the [=storage quota=], [=/reject=] |p| with a {{QuotaExceededError}} and abort,
+         leaving |stream|.[=[[buffer]]=] unmodified.
+
+         Note: [=Storage quota=] only applies to files stored in the [=origin private file system=].
+         However this operation could still fail for other files, for example if the disk being written
+         to runs out of disk space.
```
Comment (a-sully) [inline index.bs:1008]: Should we link to the new "Filling up a users disk" section here? (and below at line 1030)

File: index.bs [hunk 2]
```diff
@@ -1748,3 +1762,24 @@ Another risk factor is that of ransomware attacks. The limitations described abo
...
+Without this API websites can write data to disk not subject to quota limitations already
+by triggering downloads of large files (potentially created client side, to not incur any network
+overhead). While the presence of {{FileSystemWritableFileStream/truncate()}} and writing at a
+potentially really large offset past the end of a file makes it much easier and lower cost to
+create large files, on most file systems such files should not actually take up as much disk space as
+most commonly used file systems support sparse files (and thus wouldn't actually store the NUL
+bytes generated by resizing a file or seeking past the end of it).
+
+Whatever mitigations user agents use to guard against websites filling up a disk via either
+quota managed storage, or the existing downloads mechanism should also be employed when websites
```
Comment (a-sully) [inline index.bs:1784]:
```suggestion
quota managed storage or the existing downloads mechanism should also be employed when websites
```

**Review by a-sully — APPROVED** [review]
LGTM % nits

File: index.bs (hunk 1 above)
Comment (mkruisselbrink) [reply index.bs:1008]: Not sure, they are kind of separate (but related) things, I don't think linking there from here really adds much.

**Pull request merged**